src/gantry/state.py

```
@@ -39,3 +40,39 @@ def save_state(context: str, namespace: str) -> None:
         logger.debug(f"Could not save state: {e}")
+
+
+def load_filter_presets() -> dict:
+    """Load saved filter presets. Returns {name: expr} dict."""
+    try:
+        if _PRESETS_FILE.exists():
+            data = json.loads(_PRESETS_FILE.read_text())
+            if isinstance(data, dict):
+                return data
+    except (OSError, UnicodeDecodeError, json.JSONDecodeError) as e:
+        logger.debug(f"Could not load filter presets: {e}")
```

Ruff (0.15.13)

[warning] Logging statement uses f-string

(G004)